src/framework/jobState/FileSystemJobState/FileSystemJobState.ts

```diff
@@ -45,16 +50,36 @@ export class FileSystemJobState implements JobState {
     filter: GraphObjectFilter,
     iteratee: GraphObjectIteratee<Entity>,
   ) {
-    /* stub for now */
-    return;
+    const entitiesMatchingType = this.entities.filter(
+      (e) => e._type === filter._type,
+    );
+    for (const entity of entitiesMatchingType) {
+      await iteratee(entity);
+    }
+
+    await iterateEntityTypeIndex({
+      cacheDirectory: this.cacheDirectory,
+      type: filter._type,
+      iteratee,
+    });
```

Comment:
Is this meant to iterate things that have not been flushed and things that have been flushed? What do you think about ensuring that all things are flushed, then just walk what is on disk?

Reply:
yeah, this would make the code a little more simple. Changing